File: Sorting/41.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var firstMissingPositive = function(nums) {
  let i = 0;

  while(i < nums.length)  
  {
    let currNum = nums[i];
    if(currNum > 0 && currNum - 1 < nums.length && currNum !== nums[currNum - 1]){
      //swap
      let temp = nums[currNum - 1];
      nums[currNum - 1] = nums[i];
      nums[i] = temp;
    } else {
      i++;
    }
  }

  for(let j = 0; j < nums.length; j++){
    if(nums[j] != j + 1){
      return j + 1; 
    }
  }

  return nums.length + 1;
};

File: Sorting/41.cs
public class Solution {
    public int FirstMissingPositive(int[] nums) {
      int i = 0;

      while(i < nums.Length)
      {
        int currNum = nums[i];
        if(currNum > 0 && currNum - 1 < nums.Length && currNum != nums[currNum - 1])
        {
          //swap 
          int temp = nums[currNum - 1];
          nums[currNum - 1] = currNum;
          nums[i] = temp;
        }
        else
        {
          i++; 
        }
      }

      for(int j = 0; j < nums.Length; j++)
      {
        if(nums[j] != j + 1) return j + 1;
      }

      return nums.Length + 1;
    }
}